### PR TITLE
Simplify untyped-defs config in mypy.ini

### PR DESCRIPTION
**Pull request by DMRobertson**
Bask in _the glory_

Arguably:

Closes #11302.

#11315 remains, but that was somewhat contravertial and has now bitrotted.

### CODE REVIEW

File: mypy.ini [hunk 1]
```diff
@@ -86,194 +87,36 @@ exclude = (?x)
...
+[mypy-synapse.server]
+disallow_untyped_defs = False
 
-[mypy-tests.federation.transport.test_client]
-disallow_untyped_defs = True
+[mypy-synapse.storage.database]
+disallow_untyped_defs = False
 
+[mypy-tests.*]
+disallow_untyped_defs = False
```
[the remark this answers is not in the document]
Comment (DMRobertson) [reply mypy.ini:119]: Yeah, I was just in the process of restoring this.

[the remark this answers is not in the document]
Comment (clokep) [reply mypy.ini:119]: Sorry to jump the gun. 👼 Just SO EXCITED TO SEE THIS!

[the remark this answers is not in the document]
Comment (DMRobertson) [reply mypy.ini:119]: Likewise! (That's why I got carried away and deleted the config for tests without thinking!)

File: mypy.ini [hunk 2]
```diff
@@ -86,177 +87,35 @@ exclude = (?x)
...
 
-[mypy-synapse.metrics.*]
-disallow_untyped_defs = True
+[mypy-synapse.logging.scopecontextmanager]
+disallow_untyped_defs = False
 
 [mypy-synapse.metrics._reactor_metrics]
-# This module imports select.epoll. That exists on Linux, but doesn't on macOS.
-# See https://github.com/matrix-org/synapse/pull/11771.
```
Comment (clokep) [inline mypy.ini:157]: Are we OK removing this comment?

**Review by clokep — APPROVED** [review]

File: mypy.ini (hunk 2 above)
Comment (DMRobertson) [reply mypy.ini:157]: Ooooooops

**Comment by DMRobertson** [comment]
Assuming the sytest failure is a flake, https://github.com/matrix-org/sytest/issues/1136

**Pull request merged**